File: src/python/euroLang.py
class EuroLang:  # Parent Class
    def __init__(self, name, branch):
        self.family = "Indo-European"
        self.name = name
        self.branch = branch


class Spanish(EuroLang):  # Child Class
    def __init__(self):
        super().__init__("Spanish", "Romance")


class Hindi(EuroLang):  # Child Class
    def __init__(self):
        super().__init__("Hindi", "Indic")


if __name__ == "__main__":
    spanish_obj = Spanish()
    hindi_obj = Hindi()

    print(f"{spanish_obj.name} is in the {spanish_obj.family} family")
    print(f"The sub-family is {spanish_obj.branch}")

    print(f"{hindi_obj.name} is in the {hindi_obj.family} family")
    print(f"The sub-family is {hindi_obj.branch}")

File: src/euroLang.java
class euroLang { //Parent Class

    protected String family = "Indo-Euro"; //Inherited to child class via 'protected'

}

class Spanish extends euroLang {

    String name = "Spanish";
    String sub_family = "Romance";

    public static void main(String[] args) {
        Spanish LangObj = new Spanish(); //Constructor
        System.out.println(LangObj.name + " is in the " + LangObj.family + " family");
        System.out.println("The sub-family is " + LangObj.sub_family);
    }

}


